src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -659,36 +659,55 @@ public StringBuilder Append(char value, int repeatCount)
                 return this;
             }
 
-            // this is where we can check if the repeatCount will put us over m_MaxCapacity
-            // We are doing the check here to prevent the corruption of the StringBuilder.
-            int newLength = Length + repeatCount;
-            if (newLength > m_MaxCapacity || newLength < repeatCount)
+            char[] chunkChars = m_ChunkChars;
+            int chunkLength = m_ChunkLength;
+
+            if (((uint)chunkLength + (uint)repeatCount) <= (uint)chunkChars.Length)
```

Comment:
FYI I'm aware I could have used the Span<>Slice trick here to get rid of another cmp on x64
```if (((nuint)(uint)chunkLength + (nuint)(uint)repeatCount) <= (nuint)(uint)chunkChars.Length)```
Reason why I didn't add it is because it's a tricky optimization for the reader to understand, and its dependent on a minor detail of AsSpan, so I didnt think its worth the trouble for this particular method

Reply:
```suggestion
#if TARGET_64BIT
            // See comment in Span<T>.Slice for how this works.
            if ((ulong)(uint)chunkLength + (ulong)(uint)repeatCount <= (ulong)(uint)chunkChars.Length)
#else
            if ((uint)chunkLength <= (uint)chunkChars.Length && (uint)repeatCount <= (uint)(chunkChars.Length - chunkLength))
#endif
```

In `System.Private.CoreLib`, it's reasonable and common to depend on this particular implementation detail, see [examples](https://github.com/search?q=repo%3Adotnet%2Fruntime%20%22%2F%2F%20See%20comment%20in%20Span%3CT%3E.Slice%20for%20how%20this%20works.%22&type=code).